script/release-notes/index.js

```diff
@@ -1,473 +1,139 @@
-    return null
-  }
-  const title = commitInfo.split('title="')[1].split('"')[0]
-  if (!title.startsWith(TITLE_PREFIX)) {
-    console.warn(`WARNING: Unknown PR title for commit "${commitHash}" in PR "${prIdent}"`)
-    return null
+  if (!semver.valid(current)) {
+    // since we don't seem to be on a stabilization branch right now,
+    // pick a placeholder name that will yield the newest branch
+    // as a comparison point.
+    current = 'v999.999.999'
   }
-  return {
-    mergedFrom: prIdent,
-    prTitle: entities.decode(title.substr(TITLE_PREFIX.length))
-  }
-}
 
-const doWork = async (items, fn, concurrent = 5) => {
-  const results = []
-  const toUse = [].concat(items)
-  let i = 1
-  const doBit = async () => {
-    if (toUse.length === 0) return
-    console.log(`Running ${i}/${items.length}`)
-    i += 1
-
-    const item = toUse.pop()
-    const index = toUse.length
-    results[index] = await fn(item)
-    await doBit()
-  }
-  const bits = []
-  for (let i = 0; i < concurrent; i += 1) {
-    bits.push(doBit())
+  let previous = null
+  for (const b of stabilizationBranches) {
+    if (semver.gte(semverify(b), semverify(current))) continue
+    if (previous && semver.lte(semverify(b), semverify(previous))) continue
+    previous = b
   }
-  await Promise.all(bits)
-  return results
+  return previous
 }
 
-const notes = new Map()
-
-const NoteType = {
-  FIX: 'fix',
-  FEATURE: 'feature',
-  BREAKING_CHANGE: 'breaking-change',
-  DOCUMENTATION: 'doc',
-  OTHER: 'other',
-  UNKNOWN: 'unknown'
-}
+const getPreviousPoint = async (point) => {
+  const currentBranch = await getBranchOf(point)
+  const currentTag = (await getTagsOf(point)).filter(t => tagIsSupported(t)).pop()
+  const currentIsStable = tagIsStable(currentTag)
+  // console.log({currentBranch, currentTag, currentIsStable})
 
-class Note {
-  constructor (trueTitle, prNumber, ignoreIfInVersion) {
-    // Self bindings
-    this.guessType = this.guessType.bind(this)
-    this.fetchPrInfo = this.fetchPrInfo.bind(this)
-    this._getPr = this._getPr.bind(this)
-
-    if (!trueTitle.trim()) console.error(prNumber)
-
-    this._ignoreIfInVersion = ignoreIfInVersion
-    this.reverted = false
-    if (notes.has(trueTitle)) {
-      console.warn(`Duplicate PR trueTitle: "${trueTitle}", "${prNumber}" this might cause weird reversions (this would be RARE)`)
-    }
-
-    // Memoize
-    notes.set(trueTitle, this)
-
-    this.originalTitle = trueTitle
-    this.title = trueTitle
-    this.prNumber = prNumber
-    this.stripColon = true
-    if (this.guessType() !== NoteType.UNKNOWN && this.stripColon) {
-      this.title = trueTitle.split(':').slice(1).join(':').trim()
-    }
+  try {
+    // First see if there's an earlier tag on the same branch
+    // that can serve as a reference point.
+    let tags = (await getTagsOnBranch(`${point}^`)).filter(t => tagIsSupported(t))
+    if (currentIsStable) tags = tags.filter(t => tagIsStable(t))
```

Comment:
I prefer to always use brackets with if statements to provide better visual context and for future expansion of logic inside the if statement

Reply:
Fixed 701a6cac2234844d388986480536354f9e4927f4

As an aside, I temporarily added `curly` to eslintrc to check for this